### PR TITLE
Editorial updates

### LINKED ISSUE #157: Integer endianness
As noted by @mikelodder7,  in the document we don't yet discuss the integers (i.e., scalars mod r) serialization. More specifically, since we have a description for the byte length of those integers (`octet_scalar_length`), what is missing is the byte order.

In general i thing little-endian is preferred (it is mostly used in bls libs like the rust crate and blst) but an issue is that I2OSP uses big-endian if im not mistaken.

There are 3 options i can think of;
1. We describe them as 32 bytes (or `octet_scalar_length` in general) in little-endian, with a note for I2OSP.
2. We describe them in a way similar to some libs (i.e., four 64 bit parts in little-endian order), but i don't think something like that is suitable. 
3. We leave it up for implementers with some notes for guidance.

At the moment I'm leaning more towards option 3. Implementations may elect to use any kind of optimization and this will allow for that. 

Also for what i know of, other specs like ecdsa and eddsa elect to define the integer's endianness but specs like bls-sigs and h2c adopt option 3, so it does not seem to be any "absolutely correct" way to do that.

### PR DESCRIPTION
**Pull request by BasileiosKal**
Closes #157, #76,  #234 

Batching together multiple small editorial updates.

- Adds some clarification on the endianness of octet strings (in the test vectors mainly).
- Adds some text to clarify the mapping of messages to scalars.
- Removes the utf8() operation and describes Ciphersuite_id and DSTs as octet strings directly.

### CODE REVIEW

File: draft-irtf-cfrg-bbs-signatures.md [hunk 1]
```diff
@@ -161,10 +161,10 @@ dst
 : The domain separation tag.
 
 I2OSP
-: As defined by Section 4 of [@!RFC8017]
+: Transform an non-negative integer to an octet string, defined in Section 4 of [@!RFC8017]. Note, the output of this operation is in big-endian order.
```
Comment (tplooker) [inline draft-irtf-cfrg-bbs-signatures.md:164]:
```suggestion
: An operation that transforms a non-negative integer into an octet string, defined in Section 4 of [@!RFC8017]. Note, the output of this operation is in big-endian order.
```

File: draft-irtf-cfrg-bbs-signatures.md [hunk 2]
```diff
@@ -161,10 +161,10 @@ dst
...
 
 OS2IP
-: As defined by Section 4 of [@!RFC8017].
+: Transform an octet string to an non-negative integer, defined in Section 4 of [@!RFC8017]. Note, the input of this operation must be in big-endian order.
```
Comment (tplooker) [inline draft-irtf-cfrg-bbs-signatures.md:167]:
```suggestion
: An operation that transforms a octet string into an non-negative integer, defined in Section 4 of [@!RFC8017]. Note, the input of this operation must be in big-endian order.
```

File: draft-irtf-cfrg-bbs-signatures.md [hunk 3]
```diff
@@ -675,11 +677,15 @@ Parameters:
 Definitions:
 
 - seed_dst, octet string representing the domain separation tag:
-            utf8(ciphersuite_id || "SIG_GENERATOR_SEED_"), where
-            ciphersuite_id is defined by the ciphersuite.
+            ciphersuite_id || "SIG_GENERATOR_SEED_," where
```
Comment (BasileiosKal) [inline draft-irtf-cfrg-bbs-signatures.md:680]:
```suggestion
            ciphersuite_id || "SIG_GENERATOR_SEED_" where
```

File: draft-irtf-cfrg-bbs-signatures.md [hunk 4]
```diff
@@ -262,7 +259,11 @@ The schemes operations defined in this section depend on the following parameter
...
 
-### Messages and generators
+### Messages
+
+Each of the core operations of the BBS signature scheme, expect the inputted messages to be scalar values within a given range (specifically 1 and r-1, where r is the prime order of the G1 and G2 subgroups, defined by each ciphersuite, see [Notation](#notation)). There are multiple ways to transform a message from an octet string to a scalar value. This document defines the `MapMessageToScalarAsHash` operation, which hashes an octet string to a scalar (see [MapMessageToScalarAsHash](#mapmessagetoscalarashash)). An application can use a different `MapMessageToScalar` operation, but it MUST be clearly and unambiguously defined, for all parties involved. Before using the core operations, all messages MUST be mapped to their respective scalars using the same operation. The defined [MapMessageToScalarAsHash](#mapmessagetoscalarashash) is the RECOMMENDED way of mapping octet strings to scalar values.
```
Comment (andrewwhitehead) [inline draft-irtf-cfrg-bbs-signatures.md:264]:
```suggestion
Each of the core operations of the BBS signature scheme expect the inputted messages to be scalar values within a given range (specifically 1 and r-1, where r is the prime order of the G1 and G2 subgroups, defined by each ciphersuite, see [Notation](#notation)). There are multiple ways to transform a message from an octet string to a scalar value. This document defines the `MapMessageToScalarAsHash` operation, which hashes an octet string to a scalar (see [MapMessageToScalarAsHash](#mapmessagetoscalarashash)). An application can use a different `MapMessageToScalar` operation, but it MUST be clearly and unambiguously defined, for all parties involved. Before using the core operations, all messages MUST be mapped to their respective scalars using the same operation. The defined [MapMessageToScalarAsHash](#mapmessagetoscalarashash) is the RECOMMENDED way of mapping octet strings to scalar values.
```

File: draft-irtf-cfrg-bbs-signatures.md [hunk 5]
```diff
@@ -1254,7 +1259,7 @@ Note that these two ciphersuites differ only in the hash function (SHAKE-256 vs
 
 **Generator parameters**:
 
-- generator\_seed: A global seed value of utf8("BBS\_BLS12381G1\_XOF:SHAKE-256\_SSWU\_RO\_MESSAGE\_GENERATOR\_SEED") which is used by the [create_generators](#generator-point-computation) operation to compute the required set of message generators.
+- generator\_seed: A global seed value of "BBS\_BLS12381G1\_XOF:SHAKE-256\_SSWU\_RO\_MESSAGE\_GENERATOR\_SEED" (i.e., an ASCII string comprising of 59 bytes) which is used by the [create_generators](#generator-point-computation) operation to compute the required set of message generators.
```
Comment (andrewwhitehead) [inline draft-irtf-cfrg-bbs-signatures.md:1262]:
```suggestion
- generator\_seed: A global seed value of "BBS\_BLS12381G1\_XOF:SHAKE-256\_SSWU\_RO\_MESSAGE\_GENERATOR\_SEED" (i.e., an ASCII string comprised of 59 bytes) which is used by the [create_generators](#generator-point-computation) operation to compute the required set of message generators.
```

File: draft-irtf-cfrg-bbs-signatures.md [hunk 6]
```diff
@@ -1288,16 +1293,16 @@ Note that these two ciphersuites differ only in the hash function (SHAKE-256 vs
 
 **Generator parameters**:
 
-- generator\_seed: A global seed value of utf8("BBS\_BLS12381G1\_XMD:SHA-256\_SSWU\_RO\_MESSAGE\_GENERATOR\_SEED") which is used by the [create_generators](#generator-point-computation) operation to compute the required set of message generators.
+- generator\_seed: A global seed value of "BBS\_BLS12381G1\_XMD:SHA-256\_SSWU\_RO\_MESSAGE\_GENERATOR\_SEED" (i.e., an ASCII string comprising of 57 bytes) which is used by the [create_generators](#generator-point-computation) operation to compute the required set of message generators.
```
Comment (andrewwhitehead) [inline draft-irtf-cfrg-bbs-signatures.md:1296]:
```suggestion
- generator\_seed: A global seed value of "BBS\_BLS12381G1\_XMD:SHA-256\_SSWU\_RO\_MESSAGE\_GENERATOR\_SEED" (i.e., an ASCII string comprised of 57 bytes) which is used by the [create_generators](#generator-point-computation) operation to compute the required set of message generators.
```

File: draft-irtf-cfrg-bbs-signatures.md [hunk 7]
```diff
@@ -1288,16 +1293,16 @@ Note that these two ciphersuites differ only in the hash function (SHAKE-256 vs
...
 
 
 # Test Vectors
 
 The following section details a basic set of test vectors that can be used to confirm an implementations correctness
 
-**NOTE** All binary data below is represented as octet strings encoded in hexadecimal format
+**NOTE** All binary data below are represented as octet strings in big endian order, encoded in hexadecimal format.
```
Comment (tplooker) [inline draft-irtf-cfrg-bbs-signatures.md:1303]:
```suggestion
**NOTE** All binary data below is represented as octet strings in big endian order, encoded in hexadecimal format.
```

File: draft-irtf-cfrg-bbs-signatures.md [hunk 8]
```diff
@@ -675,11 +677,15 @@ Parameters:
 Definitions:
 
 - seed_dst, octet string representing the domain separation tag:
-            utf8(ciphersuite_id || "SIG_GENERATOR_SEED_"), where
-            ciphersuite_id is defined by the ciphersuite.
+            ciphersuite_id || "SIG_GENERATOR_SEED_" where
+            ciphersuite_id is defined by the ciphersuite and
+            "SIG_GENERATOR_SEED_" is an ASCII string comprising of 19
```
Comment (BasileiosKal) [inline draft-irtf-cfrg-bbs-signatures.md:682]:
```suggestion
            "SIG_GENERATOR_SEED_" is an ASCII string comprised of 19
```

File: draft-irtf-cfrg-bbs-signatures.md [hunk 9]
```diff
@@ -675,11 +677,15 @@ Parameters:
...
+            bytes.
 - generator_dst, octet string representing the domain separation tag:
-                 utf8(ciphersuite_id || "SIG_GENERATOR_DST_"), where
-                 ciphersuite_id is defined by the ciphersuite.
+                 ciphersuite_id || "SIG_GENERATOR_DST_", where
+                 ciphersuite_id is defined by the ciphersuite and
+                 "SIG_GENERATOR_DST_" is an ASCII string comprising of
```
Comment (BasileiosKal) [inline draft-irtf-cfrg-bbs-signatures.md:687]:
```suggestion
                 "SIG_GENERATOR_DST_" is an ASCII string comprised of
```

File: draft-irtf-cfrg-bbs-signatures.md [hunk 10]
```diff
@@ -708,7 +714,7 @@ There are multiple ways in which messages can be mapped to their respective scal
 
 ### MapMessageToScalarAsHash
 
-This operation takes an input message and maps it to a scalar value via a cryptographic hash function for the given curve. The operation takes also as an optional input a domain separation tag (dst). If a dst is not supplied, its value MUST default to the octet string returned from utf8(ciphersuite\_id || "MAP\_MSG\_TO\_SCALAR\_AS\_HASH\_"), where ciphersuite\_id is the ASCII string representing the unique ID of the ciphersuite.
+This operation takes an input message and maps it to a scalar value via a cryptographic hash function for the given curve. The operation takes also as an optional input a domain separation tag (dst). If a dst is not supplied, its value MUST default to the octet string returned from ciphersuite\_id || "MAP\_MSG\_TO\_SCALAR\_AS\_HASH\_", where ciphersuite\_id is the ASCII string representing the unique ID of the ciphersuite "MAP\_MSG\_TO\_SCALAR\_AS\_HASH\_" is an ASCII string comprising of 26 bytes.
```
Comment (BasileiosKal) [inline draft-irtf-cfrg-bbs-signatures.md:717]:
```suggestion
This operation takes an input message and maps it to a scalar value via a cryptographic hash function for the given curve. The operation takes also as an optional input a domain separation tag (dst). If a dst is not supplied, its value MUST default to the octet string returned from ciphersuite\_id || "MAP\_MSG\_TO\_SCALAR\_AS\_HASH\_", where ciphersuite\_id is the ASCII string representing the unique ID of the ciphersuite "MAP\_MSG\_TO\_SCALAR\_AS\_HASH\_" is an ASCII string comprised of 26 bytes.
```

File: draft-irtf-cfrg-bbs-signatures.md [hunk 11]
```diff
@@ -739,7 +745,7 @@ This operation describes how to hash an arbitrary octet string to `n` scalar val
...
 
-The operation takes as input an octet string representing the message to hash (msg), the number of the scalars to return (count) as well as an optional domain separation tag (dst). If a dst is not supplied, its value MUST default to the octet string returned from utf8(ciphersuit\_id || "H2S\_"), where ciphersuite\_id is the ASCII string representing the unique ID of the ciphersuite.
+The operation takes as input an octet string representing the message to hash (msg), the number of the scalars to return (count) as well as an optional domain separation tag (dst). If a dst is not supplied, its value MUST default to the octet string returned from ciphersuit\_id || "H2S\_", where ciphersuite\_id is the octet string representing the unique ID of the ciphersuite and "H2S_" is an ASCII string comprising of 4 bytes.
```
Comment (BasileiosKal) [inline draft-irtf-cfrg-bbs-signatures.md:748]:
```suggestion
The operation takes as input an octet string representing the message to hash (msg), the number of the scalars to return (count) as well as an optional domain separation tag (dst). If a dst is not supplied, its value MUST default to the octet string returned from ciphersuit\_id || "H2S\_", where ciphersuite\_id is the octet string representing the unique ID of the ciphersuite and "H2S_" is an ASCII string comprised of 4 bytes.
```

**Review by tplooker — APPROVED** [review]

**Comment by BasileiosKal** [comment]
Discussed on WG call on 14th of Feb. Will open issue on map-message-to-scalar flexibility and on ways to enforce the mapping procedure and address it to a different PR.

**Review by andrewwhitehead — APPROVED** [review]

File: draft-irtf-cfrg-bbs-signatures.md [hunk 12]
```diff
@@ -275,9 +276,10 @@ When serializing one or more values to produce an octet string, each element wil
...
 - Non-negative integers will be serialized using `I2OSP` with an output length of 8 bytes.
 - Scalars will be serialized using `I2OSP` with a constant output length defined by a particular ciphersuite.
-- ASCII strings will be transformed into octet strings using UTF-8 encoding.
 
-Variable-length octet strings will be prepended with an integer value representing the number of bytes in the string. This length is encoded to octets using `I2OSP` with an output length of 8 bytes. For example, the octet string `0x14d` is encoded as `0x0000000000000002014d`. If the length of the octet string exceeds `2^64 - 1`, the octet string must be rejected. ASCII strings are also encoded using this process by first converting to an octet string.
+Variable-length octet strings will be prepended with an integer value representing the number of bytes in the string. This length is encoded to octets using `I2OSP` with an output length of 8 bytes. For example, the octet string `0x14d` is encoded as `0x0000000000000002014d`. If the length of the octet string exceeds `2^64 - 1`, the octet string must be rejected.
```
Comment (BasileiosKal) [inline draft-irtf-cfrg-bbs-signatures.md:280]:
```suggestion
```

**Comment by BasileiosKal** [comment]
Multiple approvals. Discussed on WG call. Opened #245 to continue discussion on map message to scalar flexibility. Merging. Thanks everyone!

**Pull request merged**